backend/prompt_studio/prompt_studio_core/views.py

```diff
@@ -339,6 +349,17 @@ def create_profile_manager(self, request: HttpRequest, pk: Any = None) -> Respon
         serializer = ProfileManagerSerializer(data=request.data, context=context)
 
         serializer.is_valid(raise_exception=True)
+
+        # Check for the maximum number of profiles constraint
+        prompt_studio_tool = serializer.validated_data[
+            ProfileManagerKeys.PROMPT_STUDIO_TOOL
+        ]
+        profile_count = ProfileManager.objects.filter(
+            prompt_studio_tool=prompt_studio_tool
+        ).count()
+
+        if profile_count >= 4:
```

Comment:
making  the 4 as constant will be better.

Reply:
Sure